saw-core/src/SAWCore/Rewriter.hs

```diff
@@ -145,12 +157,15 @@ emptyMatchState = MatchState { substitution = Map.empty, constraints = [] }
 insertLookup :: Ord k => k -> a -> Map k a -> (Maybe a, Map k a)
 insertLookup k x t = Map.insertLookupWithKey (\_ a _ -> a) k x t
 
-first_order_match :: Term -> Term -> Maybe (Map DeBruijnIndex Term)
-first_order_match pat term = match pat term Map.empty
+first_order_match :: [ExtCns Term] -> Term -> Term -> Maybe (Map VarIndex Term)
+first_order_match ctxt pat term = match pat term Map.empty
   where
+    ixs :: IntSet
+    ixs = IntSet.fromList (map ecVarIndex ctxt)
```

Comment:
We pass the context along several functions as an `[ExtCns Term]` list, only to convert it to an `IntSet` here...